src/hotspot/share/opto/compile.cpp

```diff
@@ -2530,6 +2532,18 @@ void Compile::Optimize() {
     TracePhase tp(_t_macroExpand);
     print_method(PHASE_BEFORE_MACRO_EXPANSION, 3);
     PhaseMacroExpand  mex(igvn);
+    // Do not allow new macro nodes once we started to expand
```

Comment:
```suggestion
    // Do not allow new macro nodes once we start to eliminate and expand
```
I know you just moved this comment, but i think it's worth clarifying while we're at it.